### DISCUSSION TITLE
CORS header ‘Access-Control-Allow-Origin’ missing if http status code is not 200

### DISCUSSION THREAD

**Discussion by manmohan5** [fastapi/fastapi#13098] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
import http
from starlette.middleware.cors import CORSMiddleware
app = FastAPI()
app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

def GenericResponse(**kwargs):
    data = kwargs.get("data")
    status_code= kwargs.get("status_code")
    content = {
        'api_id': str(uuid.uuid4()),
        'status_code': status_code,
        'is_success':  kwargs.get("is_success"),
        'error_message':  kwargs.get("error_message"),
        'detail':  kwargs.get("detail"),
        'data':   kwargs.get("data")
        }
    return JSONResponse(content = content,status_code = status_code)

@app.get("/")
def read_root():
    return GenericResponse(
                    status_code=http.HTTPStatus.BAD_REQUEST.,
                    is_success=False,
                    error_message="bad request",
                    detail="bad request",
                    data={},
                )
@app.get("/some_route")
def read_root_1():
    return GenericResponse(
                    status_code=http.HTTPStatus.OK,
                    is_success=True,
                    error_message="ok request",
                    detail="ok request",
                    data={},
                )
```

### Description

   
 The issue I am facing is:

If Api returns **200** status code, everything works fine. if any error returns from API, like 422, 401, 400, 500 .. etc, request fails and gives below error

Origin 'http://localhost:3001/' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

The backend is deployed on a server, and I am trying to use the endpoints from my local environment.
**Server-to-server and local-to-local requests are working fine.**
 
### Operating System

Windows

### FastAPI Version

 0.115.6

### Pydantic Version

 1.10.19

### Python Version

 3.11.9

**Comment by chaitanyarahalkar** [comment]
You need to set a specific allowed origin instead of using "*" when `allow_credentials=True`. 

For example:

```
app.add_middleware(
    CORSMiddleware,
    allow_origins=["http://localhost:3001"],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)
```

Ensure the `CORSMiddleware` is added before defining any routes or handling any requests. In your code, you have it correctly placed before defining the routes so that's not a problem. Also according to CORS specifications, if you set `allow_credentials=True`, you cannot use `allow_origins=["*"]`. Browsers enforce this restriction for security reasons. If you try to use `allow_credentials=True` and `allow_origins=["*"]` together, the browser will block the request.

**Reply by manmohan5** [reply]
i tried your solution, but still i am facing same issue

**Comment by manmohan5** [comment, marked as answer]
etc/nginx/site-enabled setup was not proper, hence blocking any cors passed using editor code
